core/spring-boot/src/main/java/org/springframework/boot/logging/log4j2/Log4J2LoggingSystem.java

```diff
@@ -123,18 +119,18 @@ public Log4J2LoggingSystem(ClassLoader classLoader) {
 	protected String[] getStandardConfigLocations() {
 		List<String> locations = new ArrayList<>();
 		locations.add("log4j2-test.properties");
-		if (isClassAvailable("com.fasterxml.jackson.dataformat.yaml.YAMLParser")) {
+		if (isClassAvailable(Log4J2RuntimeHints.YAML_TREE_PARSER_V2)) {
```

Comment:
I don't like this class to know about "runtime hints" this way. Shouldn't the link be reverse, i.e. this class exposing the constant and runtime hints referring to it?

Reply:
Fixed in https://github.com/spring-projects/spring-boot/pull/46410/commits/9139fe96d5797611bf37933f4d80bdde082c9164